### PR TITLE
Test Target Reconfiguration

### PR DESCRIPTION
**Pull request by AnthonyMDev**
- Consolidates tests into targets for unit test, integration tests, performance test, and code gen tests.
- Moves integration test into new target that only runs on CI for easier local testing.
- Cleans up a few test
- Converts schemes to use XCTestPlans
- Configures CircleCI to use new test plans

### CODE REVIEW

**Review by designatednerd — COMMENTED** [review]
Looks pretty solid, just a couple of questions!

File: Sources/ApolloTestSupport/MockURLSession.swift [hunk 1]
```diff
@@ -49,14 +49,24 @@ public final class MockURLSessionClient: URLSessionClient {
...
+protocol URLSessionDataTaskMockProtocol {
+  init()
+}
+
+private final class URLSessionDataTaskMock: URLSessionDataTask, URLSessionDataTaskMockProtocol{
+
+  @available(macOS, deprecated: 10.15)
+  override init() {
+    super.init()
+  }
```
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/ApolloTestSupport/MockURLSession.swift:68]: Yeah, I'm on Big Sur. And that is exactly why I added this :)

File: Sources/ApolloTestSupport/MockWebSocket.swift [hunk 2]
```diff
@@ -1,25 +1,23 @@
 import Starscream
 import Foundation
-import ApolloTestSupport
 @testable import ApolloWebSocket
 
-class MockWebSocket: ApolloWebSocketClient {
+public class MockWebSocket: ApolloWebSocketClient {
```
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/ApolloTestSupport/MockWebSocket.swift:5]: This was moved from `ApolloWebSocketTests` to `ApolloTestSupport` so not it's being imported from another module by the tests using it. So it has to be public.

[the remark this answers is not in the document]
Comment (designatednerd) [reply Sources/ApolloTestSupport/MockWebSocket.swift:5]: goooot it, thank you

File: Sources/ApolloTestSupport/MockURLSession.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (designatednerd) [reply Sources/ApolloTestSupport/MockURLSession.swift:68]: cool - worth adding a wee comment  to clarify?

**Review by designatednerd — APPROVED** [review]

**Comment by designatednerd** [comment]
Approved from a content standpoint, obviously we have to figure out what's giving circle a sad. You may need to turn off XCPretty to get the full log barf, but i can't repro this locally either, so you may have to remote in to the running tests 😭

File: Sources/ApolloTestSupport/MockURLSession.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/ApolloTestSupport/MockURLSession.swift:68]: Done!

**Pull request merged**